File: iOS/AppFactory/AppFactory/Http/Enum/OrganizationType.swift
//
//  OrganizationType.swift
//  AppFactory
//
//  Created by Leapfrog-Software on 2018/04/09.
//  Copyright © 2018年 Leapfrog-Inc. All rights reserved.
//

enum OrganizationType {
    case unspecified
    case corporation
    case indivisual
    
    static func create(with: Int) -> OrganizationType {
        switch with {
        case 1:
            return .corporation
        case 2:
            return .indivisual
        default:
            return .unspecified
        }
    }
    
    func toValue() -> String {
        switch self {
        case .unspecified:
            return "0"
        case .corporation:
            return "1"
        case .indivisual:
            return "2"
        }
    }
    
    func toText() -> String {
        switch self {
        case .unspecified:
            return "-"
        case .corporation:
            return "法人"
        case .indivisual:
            return "個人"
        }
    }
}

File: Android/app/src/main/java/leapfrog_inc/appfactory/Http/Enum/OrganizationType.java
package leapfrog_inc.appfactory.Http.Enum;

/**
 * Created by Leapfrog-Software on 2018/04/16.
 */

public enum OrganizationType {
    unspecified,
    corporation,
    indivisual;

    public static OrganizationType create(int value) {

        switch (value) {
            case 1:
                return OrganizationType.corporation;
            case 2:
                return OrganizationType.indivisual;
            default:
                return OrganizationType.unspecified;
        }
    }

    public String toValue() {

        switch (this) {
            case corporation:
                return "1";
            case indivisual:
                return "2";
            default:
                return "0";
        }
    }

    public String toText() {

        switch (this) {
            case corporation:
                return "法人";
            case indivisual:
                return "個人";
            default:
                return "-";
        }
    }
}
